marconi-sidechain/src/Marconi/Sidechain/Api/Query/Indexers/Utxo.hs

```diff
@@ -78,6 +81,17 @@ currentSyncedBlock env = do
     Left (InvalidIndexer err) -> Left $ IndexerInternalError err
     _other -> Left $ UnexpectedQueryResult Utxo.LastSyncedBlockInfoQuery
 
+queryBlockNoAtSlotNo :: AddressUtxoIndexerEnv -> C.SlotNo -> IO C.BlockNo
```

Comment:
Why do we need to expose this query? Shouldn't the only purpose of the block number computation be to fill the `isStable field`?

Reply:
SInce I added the BlockNoSlotNoQuery in the Utxo indexer, I thought it would make more sense to put it here. So it's simply because that query is part of the Utxo indexer module. 

Then, the user of this function is the `MintBurn` query module.

Reply:
oh right, I was thinking that we were exposing it (because of the change in `Routes` as well) but it's just that the `Routes` module contains way too much stuff that should probably be elsewhere. 😬

Reply:
`Routes`? You commented on the `Utxo` module.